### PR TITLE
Add io.GitHub.chris2511.xca

### PR DESCRIPTION
**Pull request by SharkyRawr**
Hi, I made a flatpak for XCA, a x509 certificate management utility, and would like to propose adding it to the flathub repository. The upstream author has been informed and gave his blessing: https://github.com/chris2511/xca/issues/123#issuecomment-691833630

### CODE REVIEW

**Comment by nedrichards** [comment]
bot, buid io.github.chris2511.xca

**Comment by nedrichards** [comment]
bot, build io.github.chris2511.xca

**Comment by nedrichards** [comment]
Looks like you may need to disable git fsck like so: https://github.com/flathub/com.github.junrrein.PDFSlicer/commit/910b6601fc11051237bc9051629969a72db3292c#diff-d4e5aaaf31c9a82fdef6d0a4afde3d8eR167

**Comment by SharkyRawr** [comment]
Huh that's pretty weird. Added it. Thanks :)

File: io.github.chris2511.xca.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      - "STRIP=: ./configure --prefix=/app"
+      - make -j$(nproc)
+    sources:
+      - type: git
+        url: https://github.com/chris2511/xca.git
+        tag: RELEASE.2.3.0
+        commit: 79510a24c52c6678e2daf3e5dbedb0841343ecea
+        disable-fsckobjects: true
+      - type: file
+        path: "io.github.chris2511.xca.appdata.xml"
```
Comment (LeandroStanger) [inline io.github.chris2511.xca.yaml:30]: New appdata files should use metainfo instead. https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html

File: io.github.chris2511.xca.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    <id>io.github.chris2511.xca</id>
+    <name>XCA</name>
+    <summary>X Certificate and Key management</summary>
+    <metadata_license>CC0-1.0</metadata_license>
+    <project_license>LicenseRef-proprietary=https://github.com/chris2511/xca/raw/master/COPYRIGHT</project_license>
+    <provides>
+        <binary>xca</binary>
+    </provides>
+    <launchable type="desktop-id">io.github.chris2511.xca.desktop</launchable>
+    <categories>
```
Comment (barthalion) [inline io.github.chris2511.xca.appdata.xml:12]: Categories are taken from the desktop file, no need to specify them here.

File: 0002-desktop-icon-fix.patch [hunk 3]
```diff
@@ -0,0 +1,13 @@
+diff --git a/misc/xca.desktop b/misc/xca.desktop
```
Comment (barthalion) [inline 0002-desktop-icon-fix.patch:1]: Should be upstreamed.

File: io.github.chris2511.xca.appdata.xml [hunk 4]
```diff
@@ -0,0 +1,92 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (barthalion) [inline io.github.chris2511.xca.appdata.xml:1]: Should be upstreamed.

File: io.github.chris2511.xca.appdata.xml [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        <category>Qt</category>
+    </categories>    
+    <description>
+        <p>
+            XCA is intended for creating and managing X.509 certificates, certificate requests, RSA, DSA and EC private keys, Smartcards and CRLs.
+            Everything that is needed for a CA is implemented.
+            All CAs can sign sub-CAs recursively. These certificate chains are shown clearly.
+            For an easy company-wide use there are customiseable templates that can be used for certificate or request generation.
+        </p>
+        Features
```
Comment (barthalion) [inline io.github.chris2511.xca.appdata.xml:23]: I don't think you can leave text hanging like that without `<p>` tags.

File: io.github.chris2511.xca.yaml [hunk 6]
```diff
@@ -0,0 +1,68 @@
+app-id: io.github.chris2511.xca
+runtime: org.kde.Platform
+runtime-version: '5.15'
+sdk: org.kde.Sdk
+finish-args:
+  # X11 + XShm access
+  - --share=ipc
+  - --socket=x11
```
Comment (barthalion) [inline io.github.chris2511.xca.yaml:8]:
```suggestion
  - --socket=fallback-x11
```

File: io.github.chris2511.xca.yaml [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+  # Qt wants OpenGL access
+  - --device=dri
+  # Wayland access
+  - --socket=wayland
+  # Needs to save files locally
+  - --filesystem=xdg-documents
+command: /app/bin/xca
```
Comment (barthalion) [inline io.github.chris2511.xca.yaml:15]: It can be just `xca`.

File: io.github.chris2511.xca.appdata.xml [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+                </li>
+                <li>
+                    Close #163: Show key type/size on column of Certificates tab
+                </li>
+            </ul>
+        </release>
+    </releases>
+    <screenshots>
+        <screenshot type="default">
+            <image>https://www.hohnstaedt.de/xca/images/screenshots/XCA-Screenshot-certs.png</image>
```
Comment (barthalion) [inline io.github.chris2511.xca.appdata.xml:84]: I think each image needs to be in separate `<screenshot>` element.

**Review by barthalion — CHANGES_REQUESTED** [review]

**Comment by barthalion** [comment]
bot, build io.github.chris2511.xca

File: io.github.chris2511.xca.appdata.xml [hunk 9]
```diff
@@ -0,0 +1,92 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+    <id>io.github.chris2511.xca</id>
+    <name>XCA</name>
+    <summary>X Certificate and Key management</summary>
+    <metadata_license>CC0-1.0</metadata_license>
+    <project_license>LicenseRef-proprietary=https://github.com/chris2511/xca/raw/master/COPYRIGHT</project_license>
```
Comment (barthalion) [inline io.github.chris2511.xca.appdata.xml:7]: Hey @chris2511, is the license actually proprietary? Looks very similar to MIT at first glance but I wasn't sure.

**Comment by barthalion** [comment]
The build failed because an 128x128 icon is missing.

**Comment by chris2511** [comment]
Hi,

the license is generic 3-clause BSD.
Back in 2002, QT and OpenSSL came with slightly less free
licenses, so I added the remarks at the end.

Cheers

	Christian

**Comment by FakeShemp** [comment]
Shouldn't this be used as the url: https://www.hohnstaedt.de/xca/ ?

File: io.github.chris2511.xca.yaml (hunk 1 above)
Comment (SharkyRawr) [reply io.github.chris2511.xca.yaml:30]: So basically just rename xca.appdata.xml to xca.metainfo.xml ? Or are there other more subtle changes that I'm overlooking?

File: io.github.chris2511.xca.appdata.xml (hunk 2 above)
Comment (SharkyRawr) [reply io.github.chris2511.xca.appdata.xml]: Removed <categories> as requested.

File: io.github.chris2511.xca.yaml (hunk 6 above)
Comment (SharkyRawr) [reply io.github.chris2511.xca.yaml]: I changed it to use fallback-x11 as requested.

File: io.github.chris2511.xca.appdata.xml (hunk 8 above)
Comment (SharkyRawr) [reply io.github.chris2511.xca.appdata.xml:80]: Fixed.

**Comment by SharkyRawr** [comment]
> Shouldn't this be used as the url: https://www.hohnstaedt.de/xca/ ?

You're right, I removed the `index.php` bit.

File: io.github.chris2511.xca.yaml (hunk 1 above)
Comment (bilelmoussaoui) [reply io.github.chris2511.xca.yaml:30]: Yes, just the rename.

File: io.github.chris2511.xca.yaml [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+        commit: 79510a24c52c6678e2daf3e5dbedb0841343ecea
+        disable-fsckobjects: true
+      - type: file
+        path: "io.github.chris2511.xca.appdata.xml"
+      - type: patch
+        path: 0001-unset-werror.patch
+      - type: patch
+        path: 0002-desktop-icon-fix.patch
+    post-install:
+      - install -Dm644 io.github.chris2511.xca.appdata.xml /app/share/appdata/io.github.chris2511.xca.appdata.xml
```
Comment (bilelmoussaoui) [inline io.github.chris2511.xca.yaml:36]: And installing the metainfo under `/app/share/metainfo`.

File: io.github.chris2511.xca.yaml [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+finish-args:
+  # X11 + XShm access
+  - --share=ipc
+  - --socket=fallback-x11
+  # Qt wants OpenGL access
+  - --device=dri
+  # Wayland access
+  - --socket=wayland
+  # Needs to save files locally
+  - --filesystem=xdg-documents
```
Comment (bilelmoussaoui) [inline io.github.chris2511.xca.yaml:14]: Does it use QFileChooser?

Comment (SharkyRawr) [reply io.github.chris2511.xca.yaml:14]: It does use QFileDialog.

File: io.github.chris2511.xca.metainfo.xml [hunk 12]
```diff
@@ -0,0 +1,90 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+    <id>io.github.chris2511.xca</id>
+    <name>XCA</name>
+    <summary>X Certificate and Key management</summary>
+    <metadata_license>CC0-1.0</metadata_license>
+    <project_license>LicenseRef-proprietary=https://github.com/chris2511/xca/raw/master/COPYRIGHT</project_license>
```
Comment (barthalion) [inline io.github.chris2511.xca.metainfo.xml:7]: As discussed, it's not proprietary but generic 3-clause BSD.

**Review by barthalion — CHANGES_REQUESTED** [review]

**Comment by barthalion** [comment]
bot, build io.github.chris2511.xca

**Comment by barthalion** [comment]
bot, build io.github.chris2511.xca

**Pull request closed without merging**